### DISCUSSION TITLE
Task deadlock on: runWixelReader! log?

### DISCUSSION THREAD

**Discussion by bordeb** [NightscoutFoundation/xDrip#2871] [Q&A]
Hi! Latest stable version of xdrip (on Samsung Galaxy A8 (2018), Android 9) with several LibreAllHouse RPis arround the house, after changing from Libre1 to Libre2 sensor (EU sensor, direct BT connection to the sensor without MM, using OOP2 latest), on the xdrip screen I am noticing the log:
jamorham Home
XX:XX Home toast message next: xx:xx Task deadlock on: runWixelReader! @xx

![image](https://github.com/NightscoutFoundation/xDrip/assets/24623289/ed66349e-49f6-4d23-a349-b348b525c27d)

I think that it is related with LibreAllHouse connections and I am wondering why it is displayed? Is this maybe related with "Inter-app settings" > xDrip Web Service: enabled & Open Web Service: enabled, which is required for Libre2 with LibreAllHouse?

EDIT: forget to mention that everything works ok. Only this log is sometimes displayed and can not figure it out why nor if/how to disable it?

**Comment by Navid200** [comment]
@tzachi-dar

**Reply by bordeb** [reply]
Thank you very much @Navid200 for mentioning @tzachi-dar ! :-) I am fully aware that he is the guru for the answers in this field, but nevertheless, I did not want to address him directly here, as I did not want to be pushy. I was just hoping that someone more knowledgeable than me will help here! Thank you very much!! I really appreciate your time and time from other experts here in this community!

**Comment by bordeb** [comment]
After 1,5 years (with latest stable release ), I am still noticing this notification on the screen. Are there any new insights why this is happening and can this be resolved? Tnx in advance!

**Reply by Navid200** [reply]
Can you show the logs?

How often do you see this on screen?

**Reply by bordeb** [reply]
Quite often if I look thoroughly:-)
It all works well, as I already said, but it is annoying sometimes and can be confusing, especially as it is shown an the master phone that is with my child all the time.
Here is the log, filtered to see the relevant:
![20250116_212124](https://github.com/user-attachments/assets/889b5321-5048-473c-91fb-caa59e12cb76)

**Reply by Navid200** [reply]
Don't filter the logs.  I need to see everything.  I don't know what I am looking for.  But, I need to see exactly what is logged and if needed as you to enable detail logs.
Filtering logs defeats that whole intent.

**Reply by bordeb** [reply]
I am having trouble saving all the logs because xdrip reports "Could not package up all logs, using most recent". Is there a way to force larger log file locally?

**Reply by Navid200** [reply]
What is the hardware data source set to?
Can you post a screenshot of the master classic status page?

**Reply by bordeb** [reply]
Hardware source: Libre Bluetooth + wifi
![20250116_221515](https://github.com/user-attachments/assets/fdb813ac-0579-4d1f-9a3c-f05f0cc377ec)

Ss master classic status page:
![20250116_221732](https://github.com/user-attachments/assets/2babfd86-1840-41cf-a978-d670aef92a73)

**Reply by Navid200** [reply]
Do you know why Wixel is involved according to the logs?

**Reply by bordeb** [reply]
No, don't have a clue. Maybe it has something to do with LibreAllHouse covering with RPis around our house or...,?.?..
Now I just started a fresh new sensor and will see. I also just remembered that smartwatch wear integration is still active from the previous times when it was used (years ago) and now I disabled that wear integration and will see if something will change regarding this log. As soon as I get the problematic log/notification I will save the logs and post it here.

**Reply by bordeb** [reply]
This morning I have noticed the same issue even with wear integration disabled from last night so all continues

**Reply by Navid200** [reply]
I got the logs and then deleted the comment just in case the logs may contain passwords.
I will not look at any passwords.  I am looking for clues to stop xDrip from showing that log.
I will look at it today.

**Reply by bordeb** [reply]
Thank you for being so thoughtful and cautious! My trust is fully with you, please use the logs in all aspects that you think are useful. Looking forward to see your opinion. 
Tnx

**Reply by Navid200** [reply]
@bordeb When was the last time that xDrip was installed?
When you switched from Tomato to OOP 2, you did not uninstall, did you?

**Reply by bordeb** [reply]
1.) The last time the xdrip was installed was approximately a month ago on a new phone, clean install of the latest stable version and then imported saved setting from the old phone.

2.) Tomato is for MiaoMiao, am I right? Do you ask me if I made clean xdrip install when I was switching from MiaoMiao to direct Libre2 using OOP2? The swithc from MM (Libre1) to Libre 2 with OOP2 was approximately 2 years ago and in that time I did not uninstalled xdrip, only changed settings and installed oop2.

**Reply by Navid200** [reply]
If you have time, an experiment can help.
You should first do a database backup, which saves both the database as well as the settings: https://navid200.github.io/xDrip/docs/Backup-Database.html 

Then, uninstall xDrip.  Now, do a fresh install.  But, don't restore the database or the settings.
Then, set up xDrip.  
Do you still see the log?

**Reply by bordeb** [reply]
Uhh, :-) I see.
I will try but it is a master for my   kid and it will be a real effort to synchronise us all, so am not sure when it will be possible because of school etc..

Instead of uninstall/install, is it the same to just reset the xdrip app through cleaning storage and cache in the app settings?

**Reply by bordeb** [reply]
> @bordeb When was the last time that xDrip was installed? When you switched from Tomato to OOP 2, you did not uninstall, did you?

When you say Tomato, you mean the Tomato app for MM? I think I confused xdrip data source withe the app Tomato. We have never used the Tomato app, always the xdrip. Does it make any changes?

**Reply by Navid200** [reply]
It may be.  But, it is better to uninstall as long as you create a legacy backup.
After you restore, you will be back to exactly what you had before.  In other words, you can undo the experiment by restoring the backup.

This shouldn't be necessary.
The problem is that I am not a Libre user.  It will take me a long time to understand everything that is going on with respect to Libre.  
I don't have any personal experience with this part of the code.

Another approach would be to disable the master phone temporarily and replace it with a test phone with a fresh install and not restoring anything from the original phone on it.   Do the logs show up on the test phone? 

Considering I have no experience with Libre, I am not even sure if you can do this experiment with an existing sensor or if you have to do it when you start a new sensor.

Alternatively, we have to leave this and hope that someone else with more experience will help.

**Comment by Navid200** [comment]
@PhilTU 
Have you ever seen either the following toast or log?


Task deadlock on: runWixelReader!

**Reply by PhilTU** [reply]
I do not recall seeing it. I don't use the all house thing.

**Comment by bordeb** [comment]
Uhh, :-) I see.
I will try but it is a master for my   kid and it will be a real effort to
synchronise us all, so am not sure when it will be possible because of
school etc..

Instead of uninstall/install, is it the same to just reset the xdrip app
through cleaning storage and cache in the app settings?

**Comment by PhilTU** [comment]
xDrip can take over a Libre sensor any time in its life. Also can
reconnect. Necessary as can't start the sensor from xDrip
